Sutro theme

heff opened the pull request:
https://player-style-git-baobun-mux.vercel.app/themes/sutro

luwes commented:
Some issues I'm seeing:

- animations / transitions are seen on player load 
- if hovered over timedisplay, the volume bar shows unexpectedly and flickers in some cases
	I think I fixed a similar issue in vimeonova with visibility: hidden
	https://github.com/muxinc/player.style/blob/main/themes/vimeonova/template.html#L304-L320
- if volume is muted or 0% shouldn't the volume-range be empty? (no thumb)
- the hover state of the volume / mute button goes off if the cursor is not in the bottom half of the button
- the timerange preview thumb / time is quite far (too high) from the range IMO 
- the preview thumbnail border-radius seems too large to me

luwes commented on themes/sutro/template.html:
```diff
@@ -0,0 +1,692 @@
+  .media-button {
+    position: relative;
+    display: inline-block;
+    width: calc(2 * var(--base));
```
I wouldn't restrict the button width I think, that's the one that is variable. better resize on the height.

also there is a CSS var `--media-control-height` that might be handy for this.

<img width="368" alt="SCR-20240801-kumt" src="https://github.com/user-attachments/assets/19b14214-ecd8-4c55-bcbe-61fa7cba1915">

luwes commented on themes/sutro/template.html:
```diff
@@ -0,0 +1,692 @@
+      bottom: 0;
+      width: 100%;
+      height: calc(7 * var(--base));
+      background: linear-gradient(rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.6));
```
it's a bit more code but easing gradients make a big difference on light backgrounds
https://github.com/muxinc/elements/pull/885

heff commented:
> animations / transitions are seen on player load

How have you fixed this elsewhere? I'm seeing a few different possible approaches.

> if hovered over timedisplay, the volume bar shows unexpectedly and flickers in some cases
> I think I fixed a similar issue in vimeonova with visibility: hidden
https://github.com/muxinc/player.style/blob/main/themes/vimeonova/template.html#L304-L320

Cool, thanks. The complex animation is making this a degree more complicated.

> if volume is muted or 0% shouldn't the volume-range be empty? (no thumb)

The volume bar in the design has a border radius, which looks right, but I can't set a border radius on the bar directly, so I have to use the handle to do it. The handle has to have width to have a border radius, so zero volume still shows some of the bar... Another thing exposing the range parts as css parts will help with. If you have a better idea let me know.

> the hover state of the volume / mute button goes off if the cursor is not in the bottom half of the button

Yeah, another side effect of this design. The ideal solution here is the mute button could have a volume bar dropped inside of it. But otherwise I think I'd have to wrap both the the mute button and the volume range in the same element, and complicate the layout and styling some.

> the timerange preview thumb / time is quite far (too high) from the range IMO

How do you even adjust that? I'm not seeing any vars.

> the preview thumbnail border-radius seems too large to me

Fixed

luwes commented:
@heff I think for the volume range thumb the better tradeoff might be to just have a unrounded divider because Safari is showing a glitch between track and thumb and during animation. we can later add a CSS var for track border-radius or CSS parts...
